test/spec/groupCtrlSpec.js

```diff
@@ -89,6 +91,51 @@ describe("Autocomplete controller", function() {
     });
   });
 
+  describe("chatChanged", function() {
+    it('if a new chat arrives, the new user name is extracted', function() {
+
+      expect(scope.response.length).to.be.eq(0);
+      expect(scope.usernames.length).to.be.eq(0);
+
+      scope.group = {}
+      scope.group.chat = [{msg: "new chat", user: "boo"}];
+      expect(scope.response.length).to.be.eq(0);
+      expect(scope.usernames.length).to.be.eq(0);
+    })
+  });
+
+  describe("addNewUser", function() {
```

Comment:
It calls it implicitly, but I wasn't sure exactly how to test that. I will look at the example with the spy and try to update it tonight.